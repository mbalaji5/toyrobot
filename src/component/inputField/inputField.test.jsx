import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import { InputField } from ".";

describe("Should validate Error component", () => {
  it("should validate snapshot", () => {
    const { baseElement } = render(
      <InputField handlecommend={() => {}} clear={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it("should validate onClick", () => {
    const handlecommend = jest.fn();
    const clear = jest.fn();
    const wrapper = shallow(
      <InputField handlecommend={handlecommend} clear={clear} />
    );
    expect(handlecommend).toBeCalledTimes(0);
    expect(clear).toBeCalledTimes(0);
    wrapper.find("#submit").simulate("click");
    wrapper.find("#clear").simulate("click");

    expect(handlecommend).toBeCalledTimes(1);
    expect(clear).toBeCalledTimes(1);
  });
  it("should validate input field and click", () => {
    const handlecommend = jest.fn();
    const clear = jest.fn();
    const wrapper = shallow(
      <InputField handlecommend={handlecommend} clear={clear} />
    );

    wrapper
      .find("#inputCommend")
      .props()
      .onChange({ target: { value: "place 1,2,west" } });

    wrapper.find("#submit").simulate("click");

    expect(handlecommend).toBeCalledTimes(1);
    expect(handlecommend).toBeCalledWith("PLACE 1,2,WEST");
  });
});
