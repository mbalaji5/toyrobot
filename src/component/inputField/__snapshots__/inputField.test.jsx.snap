// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should validate Error component should validate snapshot 1`] = `
<body>
  <div>
    Please enter commend:
    <input
      data-testid="inputCommend"
      id="inputCommend"
      type="text"
      value=""
    />
    <button
      id="submit"
    >
      Submit
    </button>
    <button
      id="clear"
    >
      Clear
    </button>
  </div>
</body>
`;
